File: timerx-core/src/main/java/timerx/format/TimeUnitType.java
package timerx.format;

import static timerx.util.Constants.Symbols.SYMBOL_HOURS;
import static timerx.util.Constants.Symbols.SYMBOL_MINUTES;
import static timerx.util.Constants.Symbols.SYMBOL_REM_MILLIS;
import static timerx.util.Constants.Symbols.SYMBOL_SECONDS;

enum TimeUnitType {
  HOURS(SYMBOL_HOURS),
  MINUTES(SYMBOL_MINUTES),
  SECONDS(SYMBOL_SECONDS),
  R_MILLISECONDS(SYMBOL_REM_MILLIS);

  private final char value;

  TimeUnitType(char value) {
    this.value = value;
  }

  public char getValue() {
    return value;
  }
}

File: timerx-formatting/src/main/kotlin/timerx/formatting/TimeUnitType.kt
package timerx.formatting

import timerx.formatting.Constants.Symbols.SYMBOL_HOURS
import timerx.formatting.Constants.Symbols.SYMBOL_MINUTES
import timerx.formatting.Constants.Symbols.SYMBOL_REM_MILLIS
import timerx.formatting.Constants.Symbols.SYMBOL_SECONDS

enum class TimeUnitType(val value: Char) {
  HOURS(SYMBOL_HOURS),
  MINUTES(SYMBOL_MINUTES),
  SECONDS(SYMBOL_SECONDS),
  R_MILLISECONDS(SYMBOL_REM_MILLIS);
}
